sources/systemd.py

```diff
@@ -0,0 +1,24 @@
+"""Detect changes in systemd's journal"""
+
+from __future__ import absolute_import
+import select
+try:
+    from systemd import journal
+except ImportError:
+    raise Exception('You need the `python-systemd` library for Python 2.7. ' +
+                    'See: https://github.com/systemd/python-systemd/')
+
+def gen_lines():
+    """generator which yields new lines from journalctl, or None"""
+    j = journal.Reader()
+    j.seek_tail()
+    p = select.poll()
```

Comment:
Credits go to the [Systemd devs](https://github.com/systemd/python-systemd/) and the [FreeDesktop documentation](https://www.freedesktop.org/software/systemd/python-systemd/journal.html) :)